packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -220,10 +236,43 @@ const generateComponentMeta = (): void => {
       requiredProps.push(invalidLinkUsageProp);
     }
 
-    let [, rawPropTypes] = /const [a-z][a-zA-z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
-    rawPropTypes = rawPropTypes
-      ?.replace(/([a-zA-Z]+): (.+),/g, '$1: "$2",') // wrap values in quotes to make the object evaluable
-      .replace(/[^"](AllowedTypes\.(?:shape|oneOf).+\([{[][\s\S]+?[}\]]\)),/g, '`$1`,'); // wrap multiline shape object and oneOf array in backticks
+    let [, rawPropTypes] = /const [a-z][a-zA-Z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
+
+    // handle shared props
+    const sharedPropsName = rawPropTypes?.match(/(?<=\.{3})(([a-zA-Z]+)(?=,))/g);
+    if (sharedPropsName) {
+      // TODO: currently the scenario for multiple shared props is not supported
+      if (sharedPropsName.length > 1) {
+        throw new Error('Currently the scenario for multiple shared props is not supported.');
+      }
+      const sharedPropsConstName = sharedPropsName[0];
```

Comment:
Can be destructured like this.
```suggestion
      const [sharedPropsConstName] = sharedPropsName;
```